server/api/src/rsGeneralAdmin.cpp

```diff
@@ -1135,9 +1139,43 @@ _rsGeneralAdmin( rsComm_t *rsComm, generalAdminInp_t *generalAdminInp ) {
     }
     if ( strcmp( generalAdminInp->arg0, "rm" ) == 0 ) {
         if ( strcmp( generalAdminInp->arg1, "user" ) == 0 ) {
-            return irods::remove_user(*rsComm,
-                                      generalAdminInp->arg2 ? generalAdminInp->arg2 : "",
-                                      generalAdminInp->arg3 ? generalAdminInp->arg3 : "");
+            const auto* user_name = generalAdminInp->arg2 ? generalAdminInp->arg2 : "";
+            const auto* zone_name = generalAdminInp->arg3 ? generalAdminInp->arg3 : "";
+            namespace iua = irods::experimental::administration;
+            const auto user_and_zone = irods::user::validate_name(user_name);
+            // new functionality only enforced on versions >= 4.3.2
+            const auto client_version = irods::to_version(rsComm->cliVersion.relVersion);
+            constexpr irods::version v4320{4, 3, 2};
+            if (user_and_zone != std::nullopt && client_version != std::nullopt && client_version >= v4320) {
+                std::string derived_zone = std::get<1>(user_and_zone.value());
+                bool user_exists;
+                if (!derived_zone.empty()) {
+                    user_exists =
+                        iua::server::exists(*rsComm, iua::user{std::get<0>(user_and_zone.value()), derived_zone});
+                }
+                else {
+                    user_exists =
+                        iua::server::exists(*rsComm, iua::user{std::get<0>(user_and_zone.value()), zone_name});
```

Comment:
Consider isolating the branches to modifying the zone argument. That is, move the call to `exists()` out of the if-ladder and update the `derived_zone` variable.

That removes the chance for the calls to `exists()` becoming out of sync.